website/blog/2022-10-18-how-to-design-metrics.md

```diff
@@ -0,0 +1,367 @@
+
+- **Useful Dimensions:**
+    - `customer_status`: This field is helpful to end users because it allows them to break down the revenue generated by each customer status grouping. Members of the retention team might be interested in understanding the long-term trends of revenue from the Churn Risk group so that they can better understand the impact that their retention initiatives campaigns have had.
+    - `order_country`: This field is useful because it allows members of the finance team to break down the accepted revenue from each country of origin so that they can better understand which countries are experiencing the highest growth.
+- **Less Useful Dimensions:**
+    - `order_status` : Given that order_status is a part of the metric definition, it doesn’t make sense to include in the acceptable dimensions list because the value returned would always be `completed`.
+    - `order_id`: Each order id corresponds to a single order and a single point in time. Grouping the metric of revenue by order_id would just return the base grain of the table and the same value as the amount field - not useful from a metric perspective!
+- **Nuanced Dimensions:**
+    - `customer_id`: This is an interesting field because it can be both good and bad depending on the context in which it is used and the underlying data. In our example use case, this dimension wouldn’t be that useful - it would contain too many unique values and tracking the individual revenue impact by a single customer doesn’t make sense on a retail scale.
+        - In a SaaS business though, it might make more sense - especially with usage based pricing. The Customer Success team might be interested in tracking the revenue of certain customers and ensuring that they remain consistent.
+
+To quote Cameron Afzal, Product Manager of the dbt Semantic Layer:
+
+> Thoughtful curation of dimensions is essential for three main reasons:
+- ****Relevance:**** Analysts must include the dimensions most relevant to answering the question.
+- ****Trust****: Curating high-quality dimensions with little to no known errors helps ensure trust in analysis results and the decisions that follow.
+- ****Efficiency****: Curation provides a faster path to high-quality analysis results.
+> 
+
+To put it another way, **metrics are most useful when every dimension provided can help provide answers to the business.** 
+
+### Advanced example - how to design complicated metrics
+
+Now let’s look at a more complicated example of a metric that is calculated from components that could be metrics unto themselves. The metric in question is Net Promoter score, which is used by the dbt Labs internal analytics team to understand the experience that users are having on dbt Cloud.
```

Comment:
```SUGGESTION
Now let’s look at a more complex example of a metric  - one that is built from components that could theoretically themselves be metrics. The metric in question is Net Promoter score, which is used by the dbt Labs internal analytics team to understand the experience that users are having on dbt Cloud.

Reply:
Also this may or may not be too much of a throwback but 
![image](https://user-images.githubusercontent.com/8147311/194768432-66ef0941-0ee7-4751-8782-f3f575485c57.png)